sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/exemplar/ExemplarReservoir.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.sdk.metrics.exemplar;
+
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.context.Context;
+import io.opentelemetry.sdk.metrics.data.Exemplar;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * An interface for an exemplar reservoir of samples.
+ *
+ * <p>This represents a reservoir for a specific "point" of metric data.
+ */
+public interface ExemplarReservoir {
+
+  /** Offers a measurement to be sampled. */
+  public void offerMeasurementLong(long value, Attributes attributes, Context context);
+  /** Offers a measurement to be sampled. */
+  public void offerMeasurementDouble(double value, Attributes attributes, Context context);
+
+  /**
+   * Builds (an immutable) list of Exemplars for exporting from the current reservoir.
+   *
+   * <p>Additionally, clears the reservoir for the next sampling period.
+   *
+   * @param pointAttributes the {@link Attributes} associated with the metric point. {@link
+   *     Exemplar}s should filter these out of their final data state.
+   * @return An (immutable) list of sampled exemplars for this point.
+   */
+  public List<Exemplar> collectAndReset(Attributes pointAttributes);
+
+  /** An exemplar reservoir that stores no exemplars. */
+  public static ExemplarReservoir empty() {
+    return EMPTY;
+  }
+
+  static final ExemplarReservoir EMPTY =
```

Comment:
Move to some package private class to make sure not public

Reply:
Moved.